### PR TITLE
Validate EQL endpoints

### PR DESCRIPTION
**Pull request by philkra**
Validate all eql.* endpoints.

### CODE REVIEW

File: specification/specs/x_pack/eql/EqlHits.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class EqlHits<TEvent> {
+  total?: TotalHits
+  events?: EqlHitsEvent<TEvent>[]
+  sequences?: EqlHitsSequence<TEvent>[]
```
Comment (delvedor) [inline specification/specs/x_pack/eql/EqlHits.ts:23]: I think these properties are always present, so there is no need to add the `?`.

File: specification/specs/x_pack/eql/EqlHitsEvent.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class EqlHitsEvent<TEvent> {
+  _index?: IndexName
+  _id?: Id
+  _source?: TEvent
+  fields?: Dictionary<Field, UserDefinedValue[]>
```
Comment (delvedor) [inline specification/specs/x_pack/eql/EqlHitsEvent.ts:24]: Same here.

File: specification/specs/x_pack/eql/EqlHits.ts (hunk 1 above)
Comment (philkra) [reply specification/specs/x_pack/eql/EqlHits.ts:23]: `sequences` are conditional to the `sequence` keyword in a EQL query and if `sequences` is an available property, the `events` is not 😄 
This tests shows that `total` is also conditional: `xpack-Execute_EQL_events_query_with_filter_path_2_response.json.test-d.ts`

In other words all 3 properties must stay optional.

File: specification/specs/x_pack/eql/EqlHitsEvent.ts (hunk 2 above)
Comment (philkra) [reply specification/specs/x_pack/eql/EqlHitsEvent.ts:24]: No everything is optional as well, look at this test: `xpack-Execute_EQL_sequence_with_filter_path_2_response.json.test-d.ts` e.g. where only `fields` is available.

Comment (delvedor) [reply specification/specs/x_pack/eql/EqlHitsEvent.ts:24]: The reason is that the request is using `filter_path`, but in reality, those properties should be required.
This is something we should fix on the validator level.

**Review by delvedor — APPROVED** [review]
This is now good to go!

**Pull request merged**